include/exadg/operators/finite_element.h

```diff
@@ -0,0 +1,128 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_OPERATORS_FINITE_ELEMENT_H_
+#define INCLUDE_EXADG_OPERATORS_FINITE_ELEMENT_H_
+
+// deal.II
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_simplex_p.h>
+#include <deal.II/fe/fe_system.h>
+
+// ExaDG
+#include <exadg/grid/grid_data.h>
+#include <exadg/utilities/exceptions.h>
+
+namespace ExaDG
+{
+/**
+ * Creates and returns a shared pointer to dealii::FiniteElement<dim> depending on the element type,
+ * the type of function space, the number of components of the solution field, and the polynomial
+ * degree of the shape functions. Regarding the function space, the code assumes an H^1 conforming
+ * space (Continuous Galerkin) if is_dg (= L^2-conforming space, Discontinuous Galerkin) is false.
+ */
+template<int dim>
+std::shared_ptr<dealii::FiniteElement<dim>>
+create_finite_element(ElementType const & element_type,
+                      bool const          is_dg,
+                      unsigned int const  n_components,
+                      unsigned int const  degree)
+{
+  std::shared_ptr<dealii::FiniteElement<dim>> fe;
+
+  if(n_components == 1)
+  {
```

Comment:
I think technically we do not need to make the distinction between a scalar element, say `FE_Q<dim>(degree)`, and an `FESystem` with 1 component, `FESystem<dim>(FE_Q<dim>(degree), 1)`. The matrix-free evaluators will reduce this to the same code (at least I would assume so, otherwise it is a bug), so our central routines do not differ. The code working with `dealii::FEValues` will probably be somewhat more expensive because `dealii::FEValues` does some extra roundtrip for `FESystem` to copy things around, but we only use it in uncritical places (from a performance point of view) that I would be tempted to uniformly use one setting.

Reply:
Indeed, I was unsure here and did not want to introduce a performance penalty for the scalar case. If you want, I can adjust the code as you suggest.

Reply:
Yes, let's do that now.